src/tracks/graph/graph-track.ts

```diff
@@ -143,16 +152,23 @@ export default class GraphTrack extends CanvasTrack {
       options,
       plot,
       plots,
+      _transformCondition,
     } = this;
     if (!data) return;
     if (options.transform) {
-      // console.log('PREPARE DATA', this.id);
-      options.transform(data, scale).then(transformed => {
-        setPlotData(plots, transformed);
-        plot();
-      });
+      const condition = Math.round(ScaleHelper.getDomainSpan(scale, false) * 10);
+      this._transformCondition = condition;
+      if (options.alwaysTransform || !_transformCondition || _transformCondition !== condition) {
```

Comment:
should add a comment that the value `this._transformCondition` will not be the same as the value from `_transformCondition` when `this._transformCondition = condition`  and that therefore the check for `_transformCondition !== condition` is possible

Reply:
Made it more clear in commit df64aac4d6b6f3c417389a209adfc870b453f71b